documents/extension/funding/service.go

```diff
@@ -403,3 +412,72 @@ func (s service) DeriveFundingListResponse(model documents.Model) (*clientfundin
 	}
 	return response, nil
 }
+
+func (s service) createSignAttrs(model documents.Model, idxFunding string, selfDID identity.DID, account config.Account) ([]documents.Attribute, error) {
+	var attributes []documents.Attribute
+	data, err := s.deriveFundingData(model, idxFunding)
+	if err != nil {
+		return nil, err
+	}
+
+	signMsg, err := json.Marshal(data)
```

Comment:
We agreed we would like to sign all fields of the funding struct.
Currently, I sign it as a JSON.  Should we use another format?

Reply:
in the ideal case, we generate a merkle root hash of the funding fields

Reply:
I would love it if we don't have to rely on internal representations for signature payload generations even though its json.

Reply:
What alternative do you have in your mind?
Adding the definition to the protobuf repo or sorting the fields?

Reply:
yeah, something like that. That would be encoder agnostic IMO

Reply:
I was going to ask, can we generate proofs about of a custom attribute? We should be able right? JSON will not be easily verifiable onchain. One thing is verifying signature (which you need the `concat(fundingField1+fundingField2...)` or `fundingRoot`) and another verifying one single field of the fundingrequest as a proof.
I agree that we should have this document as a protobuf (otherwise we cannot create proofs), but then I am confused if it should be defined in `centrifuge-protobufs` repo or in `go-centrifuge` as it is an extension/property of another document. Thoughts?

Reply:
If we are adding it to protobuf I think we should use the protobuf repo

Reply:
@mikiquantum  I discussed with Ved, We will keep it for now as JSON but I will schedule a meeting where we discuss a long term solution for the next iteration. Maybe the end of this week or next week.